### ISSUE TITLE
Right arrow on Samsung keyboard not working

### ISSUE THREAD

**Issue by ragazzojp** [termux/termux-app#1906]
**Problem description**

Right arrow present on Samsung virtual keyboards does not work.

**Steps to reproduce**

Obtain a Samsung device with Samsung keyboard matching the version below. Older virtual keyboards don't have arrow keys. The one I'm talking about has also backspace and delete as keys, both working.

Open the terminal, default settings and packages. Type some text. Move left and right using the two arrows provided by the keyboard next to the world icon, below right shift. Move also using CTRL+arrow to jump word by word (e.g. in bash or nano). Right arrow produce no effect, left arrow does, basically you can only move left.

VolumeUp+D works. It seems to be independent of the language set.

**Expected behavior**

Right arrow works as well as left arrow, including combinations such as CTRL+right arrow.

**Additional information**

* Termux application version: 0.101 from Google Play
* Android OS version: 10
* Device model: Samsung S5e
* Samsung keyboard: 5.1.15.1

**Comment by RalfWerner** [comment]
![grafik](https://user-images.githubusercontent.com/45426380/105639075-79599d00-5e76-11eb-8d52-f872aa059817.png)
I have a corresponding keyboard (sorry in German here) with the settings in `termux/termux.properties` works _right!_

**Comment by ragazzojp** [comment]
I don't have any special setting, just the default.

Interesting to see that termux seems to be aware of this keyboard layout, as it removes left and right arrows from the shortcut bar, next to ESC tab CTRL and so on on top of the keyboard. I have for example up and down arrows, because the virtual keyboard doesn't have them, but I don't have left and right arrows since they're provided by the keyboard already.

**Comment by RalfWerner** [comment]
With Samsung KB is `enforce-char-based-input=true` required (known bug/workaround)! _**With you, too?**_

**Comment by ragazzojp** [comment]
Yes, this setting solves it, thanks.

However note that I don't experience bugs like the one `enforce-char-based-input` is supposed to solve. I didn't need to press space to see my text, meaning the virtual keyboard didn't buffer the keys.

I think the bugs are unrelated and this setting somehow resolve the bug I reported by chance. I would try to investigate and solve the issue anyway. It is always good if users are not forced to configure a setting to avoid a bug.

I can help you debugging, is there a tool to see which scan codes or characters are received by the terminal?

**Comment by ragazzojp** [comment]
For example, if I type `dd` with no parameters, I enter in a mode where special characters are printed. Pressing the keys, I see the following:

Arrow up -`^[[A`
Arrow down - `^[[B`
Arrow right - `^[[C` only when setting is enabled
Areow left - `^[[D`

No clue why right arrow doesn't work.

**Comment by RalfWerner** [comment]
The _workarround_ (and reasons for it) is described at the end of this [wiki](https://wiki.termux.com/wiki/Terminal_Settings) page. The problem has to fix  in Android and _not_ Linux and can only be solved if the developer has a Samsung device - most likely [here](https://github.com/termux/termux-app/blob/master/app/src/main/java/com/termux/app/TermuxActivity.java) - have fun with the pull request :)
Samsung has been able to add a lot (e.g. [in this article](https://www.androidpolice.com/2020/09/15/6-features-in-android-11-that-samsung-one-ui-had-first/ )) versus Android. I can understand the small group of developers if they avoid everything device-specific.
> Yes, this setting solves it, thanks.

So close this issue

**Comment by ragazzojp** [comment]
Give me please some days to debug it, I'll close if I can't find it, send a PR otherwise.

**Comment by RalfWerner** [comment]
If you're an Android expert and your device supports **PiP** (pop-up view, bubbles), you can check out my [investigations](https://github.com/RalfWerner/integrated-process/issues/2#issuecomment-733700709) and you may find solutions for _**my**_ bugs!
A closed issue is also _retained_ and a PR get a _new number._ Since I am _not_ a developer I cannot close here.

**Comment by ragazzojp** [comment]
So, I debugged it directly on the device. I think this is yet another limitation of the Samsung keyboard. It does not trigger any `onKeyPreIme` in general, and in particular it does not trigger anything at all, neither `onKeyPreIme` nor `onKeyDown` when pressing the right arrow. Left arrow triggers `onKeyDown`.

Closing, I'd say it's not a bug or anything on the android part of Termux. It's just simply the Samsung keyboard that doesn't pass the right arrow with default settings (`InputType.TYPE_NULL`) to the android app. There's nothing we can do.

As said previously, solution is to set `enforce-char-based-input` to `true`.
